website/public_adventures.py

```diff
@@ -49,32 +50,35 @@ def init(self, user):
                 continue
             public_profile = self.db.get_public_profile_settings(adventure.get('creator'))
             included[adventure["name"]] = True
-            adventures.append(
-                {
-                    "id": adventure.get("id"),
-                    "name": adventure.get("name"),
-                    "short_name": adventure.get("name"),
-                    "author": adventure.get("author", adventure["creator"]),
-                    "creator": adventure.get("creator"),
-                    "creator_public_profile": public_profile,
-                    "date": utils.localized_date_format(adventure.get("date")),
-                    "level": adventure.get("level"),
-                    "levels": adventure.get("levels"),
-                    "language": adv_lang,
-                    "cloned_times": adventure.get("cloned_times"),
-                    "tags": adv_tags,
-                    "text": adventure.get("content"),
-                    "is_teacher_adventure": True,
-                    # "content": adventure.get("content")
-                }
-            )
+
+            content = safe_format(adventure['formatted_content']
+                                  if adventure.get("formatted_content") else adventure["content"],
```

Comment:
Can we use the same `get` as you did in the other file here? Looks a lot cleaner.

```suggestion
            content = safe_format(adventure.get('formatted_content', adventure['content']),
```